frontend/src/components/header/Header.tsx

```diff
@@ -28,9 +32,11 @@ const Header: React.FC<HeaderProps> = ({isLogged}: HeaderProps) => {
     const handleCloseMenu = (event: React.MouseEvent<HTMLElement>) => {
         setAnchorElNav(null);
         setAnchorElUser(null);
-        const component = event.currentTarget.textContent ?? '';
-        const route = findRouteByPath(component);
-        if (route) {
+        const pageName = event.currentTarget.textContent ?? '';
+        const route = findRouteByPath(pageName);
+        if (pageName === Titles.LOG_OUT) {
```

Comment:
I really don't understand what you are doing here. The only thing you should do is to dispatch log out when user click button log out . That's all. Why are comparing something ? Of course this logic is not for route component. It's for log out component or where you use log out button

Reply:
`Log out` is just button, that doesn't redirect us to Log out page. so when i click any place in a page i seek what component i clicked, so when i clicked to Log out (in array `mainRoutes` doesn't contain Log out, because it's not component), i have to directly checked that if i clicked on Log out, i have to call dispatch(),